### PR TITLE
[Swift] style guide - Expand on the 'consider warnings as errors' recommendation

### PR DESCRIPTION
**Pull request by wltrup**
### TL;DR / Problem ###

The recommendation to 'consider warnings as errors' is a little weak.

### Solution ###

Expand on the 'consider warnings as errors' recommendation to **require** changing one's project's build settings so that **all** warnings are considered errors. Also add a screenshot showing how to make that change in the project settings.
 
### Screenshots ###

treat_warnings_as_errors

### CODE REVIEW

File: style-guides/swift/README.markdown [hunk 1]
```diff
@@ -54,6 +54,10 @@ The overarching goals are conciseness, readability, and simplicity.
 
 Consider warnings to be errors. This rule informs many stylistic decisions such as not to use the `++` or `--` operators, C-style for loops, or strings as selectors.
 
+You might also want to have the compiler enforce the above by changing your project settings to treat all warnings as errors so that your builds will fail if warnings are generated. Select your project file then search for "treat warnings" in the "Build Settings" tab. This can also be done per target, rather than for the entire project.
+
+![Treat warnings as errors](screens/treat_warnings_as_errors.png)
```
Comment (dominicfreeston) [inline style-guides/swift/README.markdown]: I'd possibly even be more forceful, let's make this a requirement?

**Comment by wltrup** [comment]
I've now changed the text to **require** rather than just suggest that warnings be considered errors.

**Comment by dominicfreeston** [comment]
I agree with this, but I'm wondering whether we should start moving out the project/xcode config things to the `ide-settings/xcode` folder? What do you think?

**Comment by gbasile** [comment]
we just made this change on oddschecker as well. Probably we could mention as well that if you intentionally want to supress a warning you should do that just for the lines involved and not for the entire file as recommended in some places. 

This is an example from oddschecker: 

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wincomplete-implementation"
@implementation NSURLRequest (AcceptBadCerts)
#pragma clang diagnostic pop
```

**Comment by dominicfreeston** [comment]
Nice @gbasile, we should also figure out what the syntax is for Swift :)

**Comment by wltrup** [comment]
I'm ok with moving the details of how to configure the project out from the style guide.

**Comment by dominicfreeston** [comment]
I'm going to 👍 for now, I'd rather have it somewhere - we can move it out from style guide into `ide-settings` in a future PR if needed.

**Pull request merged**